src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -0,0 +1,279 @@
+     * update actions on attribute definitions (e.g. changeAttributeName, changeAttributeLabel, etc..) for the required
+     * resource.
+     *
+     * @param oldAttributeDefinitions                       the old list of attribute definitions.
+     * @param newAttributeDefinitionsDrafts                 the new list of attribute definitions drafts.
+     * @param productTypeAttributeDefinitionActionFactory   factory responsible for building attribute definitions
+     *                                                      update actions.
+     * @return a list of attribute definitions update actions if the list of attribute definitions is not identical.
+     *         Otherwise, if the attribute definitions are identical, an empty list is returned.
+     * @throws BuildUpdateActionException in case there are attribute definitions drafts with duplicate names.
+     */
+    @Nonnull
+    private static List<UpdateAction<ProductType>>
+        buildAttributeDefinitionUpdateActionsWithNewAttributeDefinitionDrafts(
+            @Nonnull final List<AttributeDefinition> oldAttributeDefinitions,
+            @Nonnull final List<AttributeDefinitionDraft> newAttributeDefinitionsDrafts,
+            @Nonnull final ProductTypeAttributeActionFactory productTypeAttributeDefinitionActionFactory)
+            throws BuildUpdateActionException {
+
+        final HashSet<String> removedAttributesDefinitionsNames = new HashSet<>();
+
+        final Map<String, AttributeDefinition> oldAttributesDefinitionsNameMap =
+            oldAttributeDefinitions
+                .stream()
+                .collect(toMap(AttributeDefinition::getName, attributeDefinition -> attributeDefinition));
+
+        final Map<String, AttributeDefinitionDraft> newAttributesDefinitionsDraftsNameMap;
+        try {
+            newAttributesDefinitionsDraftsNameMap = newAttributeDefinitionsDrafts.stream().collect(
+                toMap(AttributeDefinitionDraft::getName, attributeDefinitionDraft -> attributeDefinitionDraft,
+                    (attributeDefinitionDraftA, attributeDefinitionDraftB) -> {
+                        throw new DuplicateNameException("Attribute definitions drafts have duplicated names. "
+                            + "Attribute definitions names are expected to be unique inside their product type.");
+                    }
+                ));
+        } catch (final DuplicateNameException exception) {
+            throw new BuildUpdateActionException(exception);
+        }
+
+        final List<UpdateAction<ProductType>> updateActions =
+            buildRemoveAttributeDefinitionOrAttributeDefinitionUpdateActions(
+                oldAttributeDefinitions,
+                removedAttributesDefinitionsNames,
+                newAttributesDefinitionsDraftsNameMap,
+                productTypeAttributeDefinitionActionFactory
+            );
+
+        updateActions.addAll(
```

Comment:
You are adding first, since there is no way to add with index (as the case with assets) correct?

Reply:
yes